### PR TITLE
SSA Time Series Forecasting Tutorial

### PR DESCRIPTION
**Pull request by luisquintanilla**
## Summary

Bike Rental Demand Forecasting using SSA.

[Preview Link](https://review.docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/time-series-demand-forecasting?branch=pr-en-us-15289)

[AB#1608214](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1608214)

### CODE REVIEW

File: docs/machine-learning/tutorials/time-series-demand-forecasting.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+        public float Year { get; set; }
+
+        public float TotalRentals { get; set; }
+    }
+    ```
+
+    The `ModelInput` class contains the following columns:
+
+    - **RentalDate** the date of the observation.
```
[the remark this answers is not in the document]
Comment (luisquintanilla) [reply docs/machine-learning/tutorials/time-series-demand-forecasting.md]: #resolved

File: docs/machine-learning/tutorials/time-series-demand-forecasting.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+Inspection of the actual and forecasted values shows the following relationships:
+
+![Actual vs Forecast Comparison](./media/time-series-demand-forecasting/forecast.png)
+
+While the forecasted values are not predicting the exact number of rentals, they provide a more narrow range of values that allows an operation to optimize their use of resources. 
+
+Congratulations! You've now successfully built a time series machine learning model to forecast bike rental demand.
+
+You can find the source code for this tutorial at the [dotnet/machinelearning-samples](https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/getting-started/Forecasting_BikeSharingDemand) repository.
```
Comment (cjgronlund) [inline docs/machine-learning/tutorials/time-series-demand-forecasting.md]:
```suggestion
You can find the source code for this tutorial at the [dotnet/machinelearning-samples](https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/getting-started/Forecasting_BikeSharingDemand) repository.

## Next steps

- [Machine learning tasks in ML.NET](../resources/tasks.md)
- [Improve model accuracy](../resources/improve-machine-learning-model-ml-net.md)

```

Comment (cjgronlund) [reply docs/machine-learning/tutorials/time-series-demand-forecasting.md]: A couple of cross-links that may be useful. We need to have x-links between articles.

**Review by cjgronlund — APPROVED** [review]
Great tutorial! Some tweaks and a recommendation for information to add. But this is good to go!

File: docs/machine-learning/tutorials/time-series-demand-forecasting.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+
+The technique for analyzing data used in this tutorial is univariate time-series analysis. Univariate time-series analysis takes a look at a single numerical observation over a period of time at specific intervals such as monthly sales. 
+
+The algorithm used in this tutorial is [Single Spectrum Analysis(SSA)](http://ssa.cf.ac.uk/zhigljavsky/pdfs/SSA/SSA_encyclopedia.pdf). SSA works by decomposing a time-series into a set of principal components. These components can be interpreted as the parts of a signal that correspond to trends, noise, seasonality, and many other factors. Then, these components are reconstructed and used to forecast values some time in the future.
+
+## Create console application
+
+1. Create a new **C# .NET Core console application** called "BikeDemandForecasting".
+1. Install **Microsoft.ML 1.4.0-preview2** NuGet Package
```
[the remark this answers is not in the document]
Comment (luisquintanilla) [reply docs/machine-learning/tutorials/time-series-demand-forecasting.md]: Microsoft.ML is the package name, the last part is the version

File: docs/machine-learning/tutorials/time-series-demand-forecasting.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+
+    For the first year, only the values in the `Year` column less than 1 are selected by setting the `upperBound` parameter to 1. Conversely, for the second year, values greater than or equal to 1 are selected by setting the `lowerBound` parameter to 1.
+
+## Define time series analysis pipeline
+
+1. Define a pipeline that uses the [SsaForecastingEstimator](xref:Microsoft.ML.Transforms.TimeSeries.SsaForecastingEstimator) to forecast values in a time-series dataset.
+
+    [!code-csharp [DefinePipeline](~/machinelearning-samples/samples/csharp/getting-started/Forecasting_BikeSharingDemand/BikeDemandForecasting/Program.cs#L36-L45)]
+
+    The `forecastingPipeline` takes 365 data points for the first year
```
[the remark this answers is not in the document]
Comment (luisquintanilla) [reply docs/machine-learning/tutorials/time-series-demand-forecasting.md]: #resolved

File: docs/machine-learning/tutorials/time-series-demand-forecasting.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (luisquintanilla) [reply docs/machine-learning/tutorials/time-series-demand-forecasting.md]: #resolved

**Pull request merged**